### PR TITLE
optimize performance. decode in user thread

### PR DESCRIPTION
**Pull request by icodening**
## What is the purpose of the change
optimize performance. decode in user thread

### CODE REVIEW

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/call/TripleMessageProducer.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        } catch (Throwable t) {
+            if (throwableConsumer != null) {
+                throwableConsumer.accept(t);
+            }
+            if (t instanceof RuntimeException) {
+                throw (RuntimeException) t;
+            }
+            if (t instanceof InvocationTargetException) {
+                Throwable targetException = ((InvocationTargetException) t).getTargetException();
+                throw new RuntimeException(targetException);
```
Comment (EarthChen) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/call/TripleMessageProducer.java:54]: Why do you need to throw at the same time as consumer?

Comment (EarthChen) [reply dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/call/TripleMessageProducer.java:54]: Why use runtime to wrap exceptions?

**Review by EarthChen — CHANGES_REQUESTED** [review]

**Review by EarthChen — APPROVED** [review]
LGTM

**Review by guohao — APPROVED** [review]
LGTM

**Pull request merged**